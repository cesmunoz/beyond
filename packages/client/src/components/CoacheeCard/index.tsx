import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import Avatar from '@material-ui/core/Avatar';
import Box from '@material-ui/core/Box';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import Typography from '@material-ui/core/Typography';

import { getInitials } from 'utils/textTransform';
import useTypedSelector from 'selectors/typedSelector';
import useStyles from './styles';

type CoacheeCardProps = {
  coacheeId: string;
  displayName: string | null;
  company?: string;
  status: string;
  selected?: boolean;
  isDetail?: boolean;
  email?: string;
  onClick?: (e: React.MouseEvent<HTMLElement>) => void;
  selectedCoachee?: string | null;
};

const CoacheeCard: React.FC<CoacheeCardProps> = ({
  coacheeId,
  displayName,
  company,
  status,
  isDetail,
  selected,
  email,
  onClick,
}) => {
  const { mobile } = useTypedSelector(state => state);
  const classes = useStyles();

  return (
    <Card
      id={coacheeId}
      className={cn({
        [classes.card]: !isDetail,
        [classes.cardDetail]: isDetail,
        [classes.selected]: selected,
      })}
      onClick={onClick}
      style={{ minHeight: company ? 70 : 64 }}
    >
      <CardContent
        className={cn(classes.cardContent, {
          [classes.cardContentDetail]: isDetail,
        })}
      >
        <Box
          display="flex"
          alignItems="center"
          justifyContent="space-between"
          className={classes.coacheeContent}
        >
          <div className={classes.avatarContainer}>
            <Avatar
              className={cn(classes.avatar, {
                [classes.avatarImage]: isDetail,
              })}
            >
              {displayName ? getInitials(displayName) : null}
            </Avatar>
          </div>
          <div className={`${classes.coacheeInfo} ${status === 'inactive' && classes.inactive}`}>
            {displayName && (
              <Typography
                title={displayName}
                variant="h6"
                className={cn(classes.coacheeDisplayName, {
                  [classes.coacheeDisplayNameDetail]: isDetail,
                })}
              >
                {displayName}
              </Typography>
            )}
            {isDetail && email && (
              <Typography
                variant="subtitle1"
                color="textSecondary"
                className={cn({
                  [classes.coacheeText]: !isDetail,
                  [classes.coacheeTextDetail]: isDetail,
                })}
                style={{ width: mobile ? 194 : '100%' }}
              >
                {email}
              </Typography>
            )}
            {company && (
              <Typography
                variant="subtitle1"
                color="textSecondary"
                className={cn({
                  [classes.coacheeText]: !isDetail,
                  [classes.coacheeTextDetail]: isDetail,
                })}
              >
                {company}
              </Typography>
            )}
          </div>
          {!isDetail && <ChevronRightIcon className={classes.coacheeChevron} />}
        </Box>
      </CardContent>
    </Card>
  );
};

CoacheeCard.propTypes = {
  coacheeId: PropTypes.string.isRequired,
  displayName: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  company: PropTypes.string,
  isDetail: PropTypes.bool,
  selected: PropTypes.bool,
  email: PropTypes.string,
  onClick: PropTypes.func,
};

CoacheeCard.defaultProps = {
  company: '',
  isDetail: false,
  selected: false,
  email: '',
  onClick: undefined,
};

export default CoacheeCard;
