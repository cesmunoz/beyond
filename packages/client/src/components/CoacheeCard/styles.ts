import { makeStyles, Theme } from '@material-ui/core/styles';
import Colors from 'styles/colors';

export default makeStyles((theme: Theme) => ({
  card: {
    alignItems: 'center',
    borderRadius: '10px',
    boxShadow: 'none',
    display: 'flex',
    marginBottom: theme.spacing(4),
    minHeight: '64px',
  },
  cardDetail: {
    borderRadius: '10px',
    boxShadow: 'none',
    display: 'flex',
  },
  cardContent: {
    overflow: 'hidden',
    width: '100%',
  },
  cardContentDetail: {
    paddingLeft: 0,
  },
  inactive: {
    color: Colors.Fonts.Gray,
  },
  avatar: {
    fontFamily: 'Muli',
    fontSize: '14px',
    fontWeight: 'normal',
    height: '32px',
    width: '32px',
  },
  avatarImage: {
    height: 96,
    width: 96,
    color: '#000',
    fontSize: '24px',
    border: 'solid 3px #ededed',
    backgroundColor: 'rgba(0, 0, 0, 0.04)',
    marginRight: '20px',
  },
  avatarContainer: {
    paddingRight: theme.spacing(1.5),
  },
  coacheeContent: {
    cursor: 'pointer',
    width: '100%',
  },
  coacheeInfo: {
    marginRight: theme.spacing(2),
    overflow: 'hidden',
    width: '90%',
  },
  coacheeDisplayName: {
    lineHeight: 1.25,
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    fontSize: '16px',
    fontFamily: 'Muli',
    fontWeight: 'bold',
  },
  coacheeText: {
    fontFamily: 'Muli',
    fontWeight: 'bold',
    lineHeight: 1.5,
    fontSize: '12px',
  },
  coacheeTextDetail: {
    color: '#000',
    fontFamily: 'Muli',
    fontWeight: 600,
    lineHeight: 1.25,
    fontSize: '16px',
    marginBottom: '4px',
    wordBreak: 'break-all',
  },
  coacheeDisplayNameDetail: {
    fontSize: '26px',
    marginBottom: '7px',
  },
  coacheeChevron: {
    color: '#000',
    fontSize: '32px',
  },
  selected: {
    border: '5px solid #30d2dd',
    boxShadow: '0 12px 20px 0 rgba(16, 132, 140, 0.2)',
  },
}));
