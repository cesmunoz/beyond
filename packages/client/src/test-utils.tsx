import React from 'react';
import { render as rtlRender, RenderOptions, RenderResult } from '@testing-library/react'; // eslint-disable-line
import { Provider } from 'react-redux';
import { RootState, configureStore } from 'store';
import APIStatus from 'constants/apiStatus';
import { ChildrenProps } from 'types';
import { COACH } from '@beyond/lib/constants';

type TestRendererOptions = {
  state: Partial<RootState>;
  renderOptions?: Omit<RenderOptions, 'queries'>;
};

const MOCK_STORE: RootState = {
  auth: {
    role: COACH,
    fullName: '',
    email: '',
    company: '',
    profileImgUploadUrl: '',
  },
  api: {
    newCoachee: APIStatus.Idle,
    newProcess: APIStatus.Idle,
    changePassword: APIStatus.Idle,
    uploadReport: APIStatus.Idle,
  },
  coachees: {},
  loading: {
    processDetail: false,
    processes: false,
    coacheeDetail: false,
    coachees: false,
    newCoachee: false,
    newProcess: false,
    questions: false,
    createAnswers: false,
    changePassword: false,
    uploadReport: false,
    updateProfile: false,
  },
  mobile: false,
  processes: {},
  questionnaires: {},
  form: {
    processId: '',
    collaborators: [],
    expectation: '',
    personalInfo: {
      birthDate: '',
      birthTime: '',
      city: '',
      company: '',
      country: '',
      education: '',
      fullName: '',
      role: '',
      seniority: '',
    },
    questionnaire: {},
  },
  summary: {
    coachees: 0,
    pendingReview: 0,
    pendingAnswers: 0,
    finished: 0,
  },
};

function render(
  ui: React.ReactElement,
  { state, ...renderOptions }: TestRendererOptions,
): RenderResult {
  const store = configureStore({ ...MOCK_STORE, ...state });

  function Wrapper({ children }: ChildrenProps): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from '@testing-library/react'; // eslint-disable-line

// override render method
export { render };
