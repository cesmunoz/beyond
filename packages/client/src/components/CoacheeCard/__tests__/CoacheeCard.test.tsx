import React from 'react';
import { RenderResult } from '@testing-library/react';

import { render } from 'test-utils';
import CoacheeCard from '..';

const renderCoachee = (status = 'active'): RenderResult =>
  render(
    <CoacheeCard displayName="falecci@morean.co" company="Morean" coacheeId="1" status={status} />,
    {
      state: {
        mobile: false,
      },
    },
  );

describe('<CoacheeCard />', () => {
  test('should render an active coachee correctly', async () => {
    const { asFragment } = renderCoachee();

    expect(asFragment()).toMatchSnapshot();
  });
  test('should render an inactive coachee correctly', async () => {
    const { asFragment } = renderCoachee('inactive');

    expect(asFragment()).toMatchSnapshot();
  });
});
