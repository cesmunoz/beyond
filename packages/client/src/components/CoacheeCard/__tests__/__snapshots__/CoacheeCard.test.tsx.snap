// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CoacheeCard /> should render an active coachee correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
    id="1"
    style="min-height: 70px;"
  >
    <div
      class="MuiCardContent-root makeStyles-cardContent-3"
    >
      <div
        class="MuiBox-root MuiBox-root-48 makeStyles-coacheeContent-9"
      >
        <div
          class="makeStyles-avatarContainer-8"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-6 MuiAvatar-colorDefault"
          >
            FA
          </div>
        </div>
        <div
          class="makeStyles-coacheeInfo-10 false"
        >
          <h6
            class="MuiTypography-root makeStyles-coacheeDisplayName-11 MuiTypography-h6"
            title="falecci@morean.co"
          >
            falecci@morean.co
          </h6>
          <h6
            class="MuiTypography-root makeStyles-coacheeText-12 MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
          >
            Morean
          </h6>
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-coacheeChevron-15"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<CoacheeCard /> should render an inactive coachee correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-card-95 MuiPaper-elevation1 MuiPaper-rounded"
    id="1"
    style="min-height: 70px;"
  >
    <div
      class="MuiCardContent-root makeStyles-cardContent-97"
    >
      <div
        class="MuiBox-root MuiBox-root-142 makeStyles-coacheeContent-103"
      >
        <div
          class="makeStyles-avatarContainer-102"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-100 MuiAvatar-colorDefault"
          >
            FA
          </div>
        </div>
        <div
          class="makeStyles-coacheeInfo-104 makeStyles-inactive-99"
        >
          <h6
            class="MuiTypography-root makeStyles-coacheeDisplayName-105 MuiTypography-h6"
            title="falecci@morean.co"
          >
            falecci@morean.co
          </h6>
          <h6
            class="MuiTypography-root makeStyles-coacheeText-106 MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
          >
            Morean
          </h6>
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-coacheeChevron-109"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
